lib/SPIRV/OCLUtil.cpp

```diff
@@ -143,10 +143,21 @@ unsigned getOCLVersion(Module *M) {
   NamedMDNode *NamedMD = M->getNamedMetadata(kSPIR2MD::OCLVer);
   if (!NamedMD)
     return 0;
-  assert (NamedMD->getNumOperands() == 1 && "Invalid SPIR");
-  MDNode *MD = NamedMD->getOperand(0);
-  unsigned Major = getMDOperandAsInt(MD, 0);
-  unsigned Minor = getMDOperandAsInt(MD, 1);
+  assert (NamedMD->getNumOperands() > 0 && "Invalid SPIR");
```

Comment:
This problem affects also other named metadata nodes that must be unique across all LLVM modules: OpenCL version, SPIR version, opencl.compiler.options & opencl.compiler.ext.options (which we are dropping for SPIR-V, but llvm-link will just concatenate all options from all the modules, which is also incorrect), opencl.enable.FP CONTRACT (must be the same for all SPIR modules).

I think the best way to fix this issue to enhance regularization pass and fix metadata duplication there. What do you think?